tests/lib/tools/store-state

```diff
@@ -0,0 +1,218 @@
+}
+
+_configure_store_backends(){
+    systemctl stop snapd.service snapd.socket
+    mkdir -p "$(dirname $STORE_CONFIG)"
+    rm -f "$STORE_CONFIG"
+    cat > "$STORE_CONFIG" <<EOF
+[Service]
+Environment=SNAPD_DEBUG=1 SNAPD_DEBUG_HTTP=7 SNAPPY_TESTING=1
+Environment=$*
+EOF
+    systemctl daemon-reload
+    systemctl start snapd.socket
+}
+
+setup_staging_store(){
+    _configure_store_backends "SNAPPY_USE_STAGING_STORE=1"
+}
+
+teardown_staging_store(){
+    systemctl stop snapd.service snapd.socket
+    rm -f "$STORE_CONFIG"
+    systemctl daemon-reload
+    systemctl start snapd.socket
+}
+
+init_fake_refreshes(){
+    local dir="$1"
+    shift
+
+    fakestore make-refreshable --dir "$dir" "$@"
+}
+
+make_snap_installable(){
+    ACK=true
+    EXTRA_DECL_JSON_FILE=""
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            (--noack) 
+                ACK=false
+                shift
+                ;;
+            (--extra-decl-json)
+                EXTRA_DECL_JSON_FILE="$2"
+                shift 2
+                ;;
+            (*)
+                break
+                ;;
+        esac
+    done
+
+    local dir="$1"
+    local snap_path="$2"
+    local snap_id="${3:-}"
+
+    if [ -n "$snap_id" ]; then
+        if ! command -v yaml2json; then
+            snap install remarshal
+        fi
+        if ! command -v jq; then
+            #shellcheck source=tests/lib/core-config.sh
+            . "$TESTSLIB"/core-config.sh
+
+            SUFFIX="$(get_test_snap_suffix)"
+            snap install "jq$SUFFIX"
+        fi
+
+        # unsquash the snap to get it's name
+        unsquashfs -d /tmp/snap-squashfs "$snap_path" meta/snap.yaml
+        snap_name=$(yaml2json < /tmp/snap-squashfs/meta/snap.yaml | jq -r .name)
+        rm -rf /tmp/snap-squashfs
+
+        cat >> /tmp/snap-decl.json << EOF
+{
+    "type": "snap-declaration",
+    "snap-id": "${snap_id}",
+    "publisher-id": "developer1",
+    "snap-name": "${snap_name}"
+}
+EOF
+
+        if [ -n "$EXTRA_DECL_JSON_FILE" ]; then
+            # then we need to combine the extra snap declaration json with the one
+            # we just wrote
+            jq -s '.[0] * .[1]' <(cat /tmp/snap-decl.json) <(cat "$EXTRA_DECL_JSON_FILE") > /tmp/snap-decl.json.tmp
+            mv /tmp/snap-decl.json.tmp /tmp/snap-decl.json
+        fi
+
+        cat >> /tmp/snap-rev.json << EOF
+{
+    "type": "snap-revision",
+    "snap-id": "${snap_id}"
+}
+EOF
+    fi
+
+    # NOTE: snap id and snap name are derived from snap file name
+    SNAP_DECL_JSON_PARAM=""
+    SNAP_REV_JSON_PARAM=""
+    if [ -n "$snap_id" ]; then
+        SNAP_DECL_JSON_PARAM="--snap-decl-json=/tmp/snap-decl.json"
+        SNAP_REV_JSON_PARAM="--snap-rev-json=/tmp/snap-rev.json"
+    fi
+    p_decl=$(fakestore new-snap-declaration --dir "$dir" "$snap_path" "$SNAP_DECL_JSON_PARAM")
+    p_rev=$(fakestore new-snap-revision --dir "$dir" "$snap_path" "$SNAP_REV_JSON_PARAM")
```

Comment:
There's a mix of lowercase and uppercase variables (which could be local as well I think). Is there a reason for that (like a convention or something)?